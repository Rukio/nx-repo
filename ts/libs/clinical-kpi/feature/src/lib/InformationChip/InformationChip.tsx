import {
  Chip,
  InfoIcon,
  Tooltip,
  makeSxStyles,
} from '@*company-data-covered*/design-system';

export enum ChipState {
  YES = 'Yes',
  NO = 'No',
}

export enum ChipActiveColor {
  RED = 'error.main',
  GREEN = 'info.main',
}

export type InformationChipProps = {
  chipState: ChipState;
  activeColor: ChipActiveColor;
  text?: string;
  dataTestId?: string;
};

const makeStyles = (activeColor: string) =>
  makeSxStyles({
    chipEnabled: {
      color: 'common.white',
      backgroundColor: activeColor,

      '&:hover': {
        cursor: 'pointer',
      },
    },
    chipDisabled: {
      color: 'common.black',
      backgroundColor: '#f0ecec',

      '&:hover': {
        cursor: 'default',
      },
    },
    chipIcon: {
      color: '#fff !important',
      opacity: 0.7,
    },
    chipPopper: {
      '& .MuiTooltip-tooltip': {
        backgroundColor: '#616161e6',
        padding: '16px 8px',
      },
      '& .MuiTooltip-arrow': {
        color: '#616161e6',
      },
    },
  });

export const InformationChip = ({
  chipState,
  activeColor,
  text,
  dataTestId = 'information-chip',
}: InformationChipProps) => {
  const styles = makeStyles(activeColor);

  const isChipEnabled = chipState === ChipState.YES;
  // NO-OP: the stubbed function is needed to alter the default location of the Chip icon
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const onDeleteStub = () => {};

  return (
    <Tooltip
      title={text}
      arrow={true}
      PopperProps={{
        sx: styles.chipPopper,
      }}
    >
      <Chip
        deleteIcon={<InfoIcon sx={styles.chipIcon} />}
        sx={isChipEnabled ? styles.chipEnabled : styles.chipDisabled}
        onDelete={isChipEnabled ? onDeleteStub : undefined}
        label={chipState}
        data-testid={dataTestId}
      />
    </Tooltip>
  );
};

export default InformationChip;
