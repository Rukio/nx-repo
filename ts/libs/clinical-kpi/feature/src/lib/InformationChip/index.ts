export { default, ChipActiveColor, ChipState } from './InformationChip';
