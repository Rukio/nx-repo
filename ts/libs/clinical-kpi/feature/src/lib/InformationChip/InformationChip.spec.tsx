import { render, screen, fireEvent } from '../util/testUtils';
import InformationChip, {
  ChipActiveColor,
  ChipState,
  InformationChipProps,
} from './InformationChip';

const setup = (props: InformationChipProps) => {
  return render(<InformationChip {...props} />);
};

describe('InformationChip', () => {
  describe('tooltip with text should be visible on mouse over', () => {
    it.each([
      {
        chipState: ChipState.YES,
        activeColor: ChipActiveColor.RED,
        text: 'test information',
      },
      {
        chipState: ChipState.YES,
        activeColor: ChipActiveColor.GREEN,
        text: 'test information',
      },
    ])(
      'renders InformationChip component correctly with active state %p and color %p with information',
      async ({ chipState, activeColor, text }) => {
        const { container } = setup({ chipState, activeColor, text });
        const button = screen.getByRole('button');
        fireEvent.mouseOver(button);
        const tooltip = await screen.findByRole('tooltip');
        expect(tooltip).toHaveTextContent(text);
        expect(container).toMatchSnapshot();
      }
    );
  });

  describe('Chip should be visible and have correct value', () => {
    it.each([
      {
        chipState: ChipState.YES,
        activeColor: ChipActiveColor.RED,
      },
      {
        chipState: ChipState.YES,
        activeColor: ChipActiveColor.GREEN,
      },
      {
        chipState: ChipState.NO,
        activeColor: ChipActiveColor.RED,
      },
      {
        chipState: ChipState.NO,
        activeColor: ChipActiveColor.GREEN,
      },
    ])(
      'renders InformationChip component correctly with active state %p and color',
      ({ chipState, activeColor }) => {
        const { container } = setup({ chipState, activeColor });
        const chip = screen.getByTestId('information-chip');
        expect(chip).toBeVisible();
        expect(chip).toHaveTextContent(chipState);
        expect(container).toMatchSnapshot();
      }
    );
  });
});
