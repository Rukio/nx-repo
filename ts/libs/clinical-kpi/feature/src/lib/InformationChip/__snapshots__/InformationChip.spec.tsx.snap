// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationChip Chip should be visible and have correct value renders InformationChip component correctly with active state {"activeColor": "error.main", "chipState": "No"} and color 1`] = `
.emotion-0 {
  max-width: 100%;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
  color: #000;
  background-color: #f0ecec;
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-0 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-0 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-0 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-0 .MuiChip-icon {
  margin-left: 5px;
  margin-right: -6px;
  color: #616161;
}

.emotion-0 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(0, 0, 0, 0.26);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-0 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-0:hover {
  cursor: default;
}

.emotion-1 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

<div>
  <div
    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault emotion-0"
    data-mui-internal-clone-element="true"
    data-testid="information-chip"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium emotion-1"
    >
      No
    </span>
  </div>
</div>
`;

exports[`InformationChip Chip should be visible and have correct value renders InformationChip component correctly with active state {"activeColor": "error.main", "chipState": "Yes"} and color 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  max-width: 100%;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
  color: #fff;
  background-color: #d32f2f;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-0 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-0 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-0 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-0 .MuiChip-icon {
  margin-left: 5px;
  margin-right: -6px;
  color: #616161;
}

.emotion-0 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(0, 0, 0, 0.26);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-0 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-0.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.2);
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-1 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  color: #fff!important;
  opacity: 0.7;
}

<div>
  <div
    class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault emotion-0"
    data-mui-internal-clone-element="true"
    data-testid="information-chip"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium emotion-1"
    >
      Yes
    </span>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault emotion-2"
      data-testid="InfoIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
  </div>
</div>
`;

exports[`InformationChip Chip should be visible and have correct value renders InformationChip component correctly with active state {"activeColor": "info.main", "chipState": "No"} and color 1`] = `
.emotion-0 {
  max-width: 100%;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
  color: #000;
  background-color: #f0ecec;
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-0 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-0 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-0 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-0 .MuiChip-icon {
  margin-left: 5px;
  margin-right: -6px;
  color: #616161;
}

.emotion-0 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(0, 0, 0, 0.26);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-0 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-0:hover {
  cursor: default;
}

.emotion-1 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

<div>
  <div
    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault emotion-0"
    data-mui-internal-clone-element="true"
    data-testid="information-chip"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium emotion-1"
    >
      No
    </span>
  </div>
</div>
`;

exports[`InformationChip Chip should be visible and have correct value renders InformationChip component correctly with active state {"activeColor": "info.main", "chipState": "Yes"} and color 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  max-width: 100%;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
  color: #fff;
  background-color: #0288d1;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-0 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-0 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-0 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-0 .MuiChip-icon {
  margin-left: 5px;
  margin-right: -6px;
  color: #616161;
}

.emotion-0 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(0, 0, 0, 0.26);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-0 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-0.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.2);
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-1 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  color: #fff!important;
  opacity: 0.7;
}

<div>
  <div
    class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault emotion-0"
    data-mui-internal-clone-element="true"
    data-testid="information-chip"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium emotion-1"
    >
      Yes
    </span>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault emotion-2"
      data-testid="InfoIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
  </div>
</div>
`;

exports[`InformationChip tooltip with text should be visible on mouse over renders InformationChip component correctly with active state {"activeColor": "error.main", "chipState": "Yes", "text": "test information"} and color %p with information 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  max-width: 100%;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
  color: #fff;
  background-color: #d32f2f;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-0 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-0 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-0 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-0 .MuiChip-icon {
  margin-left: 5px;
  margin-right: -6px;
  color: #616161;
}

.emotion-0 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(0, 0, 0, 0.26);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-0 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-0.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.2);
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-1 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  color: #fff!important;
  opacity: 0.7;
}

<div>
  <div
    aria-label="test information"
    class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault emotion-0"
    data-mui-internal-clone-element="true"
    data-testid="information-chip"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium emotion-1"
    >
      Yes
    </span>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault emotion-2"
      data-testid="InfoIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
  </div>
</div>
`;

exports[`InformationChip tooltip with text should be visible on mouse over renders InformationChip component correctly with active state {"activeColor": "info.main", "chipState": "Yes", "text": "test information"} and color %p with information 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  max-width: 100%;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
  color: #fff;
  background-color: #0288d1;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-0 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-0 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-0 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-0 .MuiChip-icon {
  margin-left: 5px;
  margin-right: -6px;
  color: #616161;
}

.emotion-0 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(0, 0, 0, 0.26);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-0 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-0.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.2);
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-1 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  color: #fff!important;
  opacity: 0.7;
}

<div>
  <div
    aria-label="test information"
    class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault emotion-0"
    data-mui-internal-clone-element="true"
    data-testid="information-chip"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium emotion-1"
    >
      Yes
    </span>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault emotion-2"
      data-testid="InfoIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
  </div>
</div>
`;
